### ISSUE TITLE
Add additional automation to the release process

### ISSUE THREAD

**Issue by jkuester** [medic/cht-core#8987]
**Describe the issue**
Our current [release process](https://github.com/medic/cht-core/blob/master/.github/ISSUE_TEMPLATE/z_release_major.md) contains a number of manual steps that are very time consuming and error prone.  We should automate as many of these steps as possible to reduce manual errors, reduce engineering time wasted on the release process, and improve the speed of the release process. All of this will be very helpful in increasing our release cadence. 

**Describe the improvement you'd like**
- [ ] Automatically verify that all commits in the release have an issue/PR included in the Milestone - could be done in the [release notes script](https://github.com/medic/cht-core/blob/master/scripts/release-notes/index.js).
- [ ] Improve handling of the Scalability Test results (many core eng do not have access to the S3 bucket). We could automatically open a PR to add the results into `tests/scalability/previous_results`.
- [ ] Updating the demo instance and pushing the latest config should be something that can be done automatically in a GitHub Action (triggered by the release build).

**Comment by jkuester** [comment]
Another thought I had based on https://github.com/medic/cht-core/pull/8989 is that the commit milestone validation logic could also be included in a CI workflow job for all PRs.  So, basically, when you open a PR, it will check the PR, any linked issues, and your PR title to see if there is a milestone configured. If not, the CI will fail for the PR.

**Comment by garethbowen** [comment]
@jkuester An alternative approach would be more like what cht-conf does which does not use milestones at all. Instead it generates the release notes from the commits and adds a comment to all issues with the release version. As we tend towards smaller and smaller releases I think this is the direction we'll ultimately go in. If that's the case the validation we need is to enforce the PR title and/or commit message complies with the common format so we can parse it programmatically.

**Comment by jkuester** [comment]
Absolutely!  FTR, we implemented a [CI check for PR titles](https://github.com/medic/cht-watchdog/blob/main/.github/workflows/conventional-commits.yml) in watchdog using `commitlint` that works well!  Would be a good place to start when looking to do something similar in cht-core, etc.

**Comment by dianabarsan** [comment]
I believe that only updating the demo instance remains. And we have really detailed instructions in the release template for that. @jkuester do you still think we should do the upgrade in a GH action, especially since it requires running helm commands against our dev cluster?

**Comment by jkuester** [comment]
IMHO, I would _much_ rather just script the helm commands in a GitHub action instead of any developer who does a release of cht-core needing AWS access.  🤷 

I have not done a ton of research into the feasibility here, but Claude does assure the that "You are totally correct" that AWS/EKS authentication in a GitHub action is pretty straightforward.... 😅   I have logged https://github.com/medic/cht-core/issues/10887 as a specific issue for these proposed changes and will close this one out as completed.
